internal/configuration/settings.go

```diff
@@ -220,32 +256,65 @@ func (s *Settings) initializeEventListeners() error {
 func (s *Settings) handleAddEvent(obj interface{}) {
 	logrus.Debug("Settings::handleAddEvent")
 
-	s.handleUpdateEvent(obj, nil)
+	if _, yes := isOurConfig(obj); yes {
+		s.handleUpdateEvent(nil, obj)
+	}
 }
 
-func (s *Settings) handleDeleteEvent(_ interface{}) {
+func (s *Settings) handleDeleteEvent(obj interface{}) {
 	logrus.Debug("Settings::handleDeleteEvent")
 
-	s.updateHosts([]string{})
+	if _, yes := isOurConfig(obj); yes {
+		s.updateHosts([]string{})
+	}
 }
 
-func (s *Settings) handleUpdateEvent(obj interface{}, _ interface{}) {
+func (s *Settings) handleUpdateEvent(_ interface{}, obj interface{}) {
 	logrus.Debug("Settings::handleUpdateEvent")
 
-	configMap, ok := obj.(*corev1.ConfigMap)
-	if !ok {
-		logrus.Errorf("Settings::handleUpdateEvent: could not convert obj to ConfigMap")
+	configMap, yes := isOurConfig(obj)
+	if !yes {
 		return
 	}
 
 	hosts, found := configMap.Data["nginx-hosts"]
-	if !found {
-		logrus.Errorf("Settings::handleUpdateEvent: nginx-hosts key not found in ConfigMap")
-		return
+	if found {
+		newHosts := s.parseHosts(hosts)
+		s.updateHosts(newHosts)
+	} else {
+		logrus.Warnf("Settings::handleUpdateEvent: nginx-hosts key not found in ConfigMap")
 	}
 
-	newHosts := s.parseHosts(hosts)
-	s.updateHosts(newHosts)
+	tlsMode, found := configMap.Data["tls-mode"]
```

Comment:
Make these Enums and validate the value is correct and figure out a way to kick the person in the head and not run in an insecure mode.